### DISCUSSION TITLE
How to fix: "The installation of this device is forbidden by system policy. Contact your system administrator."

### DISCUSSION THREAD

**Discussion by drmason789** [xdream-biking/trixter-xdream-bike#88] [Q&A] [answered]
If you have successfully set up Group Policy to prevent Windows from automatically updating to a newer driver that doesn't work (e.g. using #8), there are problems that can occur.

For example, someone or something could:
- uninstall the existing driver, then any installation of a driver for the device would be blocked.
- switch on retroactive device id restriction, then the working driver you have installed would also be blocked.

![image](https://user-images.githubusercontent.com/29954900/162636277-1771055a-98e6-4f8c-976a-5e59a6da8417.png)

What to do?

**Comment by drmason789** [comment, marked as answer]
If you are not the system administrator, you should do exactly that.
If you are, you probably need to take a look at your computer's Group Policy settings.

The instructions for keeping the PL2303 driver held back to a version that works can be found here: #8



In version 1.4.0 of diagnostic utility from this repo, there is a utility to access the group policy configuration.
Currently it supports only the 2 device IDs known to the maintainer so there's no guarantee it will correctly detect the configuration for your bike. But it's a start.

Just start the application, there's no need to connect to the bike, and click the "Driver" button.

![image](https://user-images.githubusercontent.com/29954900/162634984-cb0c46f9-8ed4-436f-a611-b751901e84c1.png)

This will read the registry to make an assessment about the Group Policy state. 

![image](https://user-images.githubusercontent.com/29954900/162635036-7cefcf55-da64-46ef-ace5-4c361a33dc28.png)

This utility doesn't change the registry, you need to do that yourself, with admin rights.

To change a group policy that is blocking your device, in summary:

- Disable the "DenyDeviceIds" feature and (maybe) restart the computer.
- Restore the old driver
- Re-enable the "DenyDeviceIds" feature

To do this:
- In regedit, navigate to HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\DeviceInstall\Restrictions
- Make a note of the values of "DenyDeviceIDs" and "DenyDeviceIDsRetroactive" in the name-values pairs on the right, then set them both to 0.
- Restart the computer
- On restart, Windows will probably try to install its current driver
- Follow the instructions in this discussion to reinstall the old driver
- In regedit, navigate to HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\DeviceInstall\Restrictions
- On the right, set "DenyDeviceIDs" to 1, and make sure "DenyDeviceIDsRetroactive" is set to 0.

For this I am assuming you have already put the device IDs under HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\DeviceInstall\Restrictions\DenyDeviceIDs
according to the instructions in #8.

**Reply by chungcu201** [reply]
Cool! The driver is all good now. I have the connection back finally :)

**Comment by drmason789** [comment]
Another option is to run X-Dream in Linux #105. 
- No PL2303 driver issue
- No blank wmv video problem (although this is solved on Windows now) #16
